### ISSUE TITLE
Simulator environment screenAPI does not handle nil parameters the same way as the game

### ISSUE THREAD

**Issue by StingerTheRaven** [nameouschangey/STORMWORKS_VSCodeExtension#27]
In VSCode, the following code:
```lua
require("_build._simulator_config")
--- default onTick function; called once per in-game tick (60 per second)

width = 0
rangeTable = {}

function onTick()
    heading = input.getNumber(3)
    range = input.getNumber(4)
    rangeTable[math.floor(heading*32)] = range*32 --heading as write index
end

--- default onDraw function; called once for each monitor connected each tick, order is not guaranteed
function onDraw()
    while (width < 32) do
        screen.drawCircle(width, rangeTable[width], 1) --attempting to use variable width as an index for rangeTable, if this is wrong, how do i do this?
        width = width + 1
    end
    if width >= 32 then
        width = 0
    end
end
```

results in the following report in the debug console:
```
c:/Users/(redacted)/.vscode/extensions/actboy168.lua-debug-1.52.0/runtime/win32-ia32/lua53/lua.exe: interrupted!
stack traceback:
	[C]: in ?
	[C]: in function 'table.concat'
	...BoatAPI\Tools\Simulator\LBSimulatorConnection.lbinternal:32: in method 'sendCommand'
	...ls/LifeBoatAPI\Tools\Simulator\LBSimulator_ScreenAPI.lua:76: in field 'drawCircle'
	...icrocontrollers/Megatank Dashboard/MyMicrocontroller.lua:31: in function 'onDraw'
	...oatAPI/Tools/LifeBoatAPI\Tools\Simulator\LBSimulator.lua:176: in method 'giveControlToMainLoop'
	...icrocontrollers\Megatank Dashboard\_build\_simulator.lua:21: in main chunk
	[C]: in ?
```

The same code, however, of course minus `require("_build._simulator_config")`, runs just fine in Stormworks proper. This image displays a perfectly functioning script on a display: https://steamuserimages-a.akamaihd.net/ugc/1866177998399745326/FD9BD222F19D4A9D5CDD3ED20156DE2B77B09574/?imw=5000&imh=5000&ima=fit&impolicy=Letterbox&imcolor=%23000000&letterbox=false

**Comment by nameouschangey** [comment]
Thanks for raising this,
It looks like the issue is you're passing in nil to screen.drawCircle (y) for the first few ticks until all 32 indices of your rangeTable actually have values.

I've renamed this bug to reflect what actually needs changed - it's that the screen.draw functions in the simulator aren't designed for handling nil values - but it seems the game doesn't have a problem with doing so.

**Comment by nameouschangey** [comment]
Fixed this, will be available in patch 0.0.8 - aiming to test and deploy in the next week.
